executor/src/witgen/vm_processor.rs

```diff
@@ -126,10 +115,13 @@ impl<'a, T: FieldElement> VmProcessor<'a, T> {
                     loop_detection_log_level = log::Level::Debug;
                 }
             }
-            if looping_period.is_none() {
-                self.compute_row(row_index, ProcessingPhase::Regular, mutable_state);
+            if looping_period.is_none() && row_index != rows_left - 1 {
```

Comment:
I don't undrestand why this new condition is not present for the loop case

Reply:
Yeah, it's a bit weird. In the normal case, we actually compute values for both the current and the next row, but in the loop case, we only set the current row. So in the normal case, we want to exit the loop one earlier. I'll add a comment to explain that.